### PR TITLE
Add data migrations with documentation

### PR DESCRIPTION
**Pull request by Gabrielpanga**
Add data migrations with documentation
Related issues: #31

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,11 +8,13 @@
...
-    "db:create:test": "npm run db:setup:test && npm run db:migrate:test up",
+    "db:create": "npm run db:setup && npm run db:migrate:schema up && npm run db:migrate:data up",
+    "db:create:test": "npm run db:setup:test  && npm run db:migrate:schema up && npm run db:migrate:data:test up",
     "db:create:all": "npm run build:tsc && npm run db:create && npm run db:create:test",
-    "db:migrate": "node ./scripts/db/migrate.js",
-    "db:migrate:test": "NODE_ENV=test npm run db:migrate",
+    "db:migrate:schema": "node ./scripts/db/schemaMigration.js",
+    "db:migrate:schema:test": "NODE_ENV=test npm run db:migrate",
+    "db:migrate:data": "node ./scripts/db/dataMigration.js",
+    "db:migrate:data:test": "NODE_ENV=test npm run db:migrate:data",
```
Comment (kaiomagalhaes) [inline package.json]: please remove this since we shouldn't use data migration for tests

File: src/db/migrations/schema/20191020.users.up.sql [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+(
+    id SERIAL PRIMARY KEY,
+    name character varying(128) NOT NULL,
+    "createdAt" TIMESTAMP
+    WITH TIME ZONE NOT NULL,
+    "updatedAt" TIMESTAMP
+    WITH TIME ZONE NOT NULL
+);
+
+    CREATE UNIQUE INDEX users_pkey_index ON users(id);
```
Comment (kaiomagalhaes) [inline src/db/migrations/schema/20191020.users.up.sql]: it seems it has an extra space here

Comment (Gabrielpanga) [reply src/db/migrations/schema/20191020.users.up.sql]: resolved

File: package.json (hunk 1 above)
Comment (Gabrielpanga) [reply package.json]: resolved

**Pull request merged**